src/uwtools/drivers/forecast.py

```diff
@@ -121,15 +129,92 @@ def requirements(self) -> None:
         ???
         """
 
-    def resources(self) -> None:
+    def resources(self, platform: dict) -> Mapping:
         """
-        ???
+        Set up the formatting for the scheduler batch script.
         """
-
-    def run(self):
+        # Add required fields to platform.
+        # Currently supporting only slurm and none.
+        slurmqueues = ["hera", "jet", "orion", "stampede"]
+        if platform["queue"] is None and platform["machine"] in slurmqueues:
+            platform["queue"] = "slurm"
+        else:
+            platform["queue"] = "none"
+
+        return {
+            "account": platform["account"],
+            "nodes": 1,
+            "queue": platform["queue"],
+            "scheduler": platform["scheduler"],
+            "tasks_per_node": 1,
+            "walltime": "00:01:00",
+        }
+
+    def run(self, log: Optional[Logger] = None) -> None:
         """
         Runs FV3.
         """
+        # Set up logging.
+        if log is None:
+            name = f"{inspect.stack()[0][3]}"
+            log = cli_helpers.setup_logging(
+                log_file="/dev/stdout",
+                log_name=name,
+                quiet=False,
+                verbose=False,
+            )
+        # Read in the config file.
+        forecast_config = config.YAMLConfig(self._config_file)
```

Comment:
Do we want to consider loading this file in `__init__()`? Sorry if I missed this discussion already.

Also, I'd suggest that we load it in like this:

```
forecast_config = config.YAMLConfig(self._config_file).get('forecast')
```

Reply:
@christinaholtNOAA Why did you choose `.get("forecast")` here instead of `["forecast"]`? If we get a `None` result, the next executable statement will fail (no?) but in a less obvious way than if this one had failed. I'm thinking that we should be able to expect correct input (it should at least have been validated against a schema) at this spot in the code, and so should be able to confidently use `[]` lookups wherever non-optional values are expected.

Reply:
Shoot. I was being sloppy with the `get("forecast")` instead of `["forecast"]`. I'm sorry for the confusion. I think we can rely on it to be there at this point in the code. 

The part that I was getting at is that we want to go ahead and load the _contents_ of the forecast section into the `forecast_config` variable and remove that top level so that we don't have to search as deep in all the other references.